### DISCUSSION TITLE
write json structure: don't want line break after "{" and "["

### DISCUSSION THREAD

**Discussion by flogawonc** [open-source-parsers/jsoncpp#1327] [General]
In my project, the json file is like:
{
	"a" : {
		"a1" : "b1",
		"a2" : [
			    14200,
			    14200,
			    14200,
			    14200
                         ]
                }
}
But when I use StreamWriter to write json file, it becomes:
{
	"a" : 
                {
		"a1" : "b1",
		"a2" : 
                         [
			    14200,
			    14200,
			    14200,
			    14200
                         ]
                }
}
I can't find a way to keep the first format of writing in the attribute of StreamWriter, so I modify the source file json_writer.cpp to solve it.

void BuiltStyledStreamWriter::writeWithIndent(String const& value) {
//to avoid "{" and "[" begin a new line.
  if (value == "{" || value == "[")        //added
    indented_ = true;                          //added
  if (!indented_)
    writeIndent();
  *sout_ << value;
  indented_ = false;
}

Hope to help someone like me.

**Comment by flogawonc** [comment]
or you can just use my fork: https://github.com/Acwme/jsoncpp

**Comment by cdunn2001** [comment]
* https://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_stream_writer_builder.html#a79bdf2e639a52f4e758c0b95bd1d3423

Our doxygen docs have some formatting problems right now, but that's trying to explain what's possible today: If you configure JSON with `indentation = ''` (empty string), then newlines and indentation should be dropped.

I don't understand what the output would be that you want. Try posting with "markdown".
